### PR TITLE
Uilist in ob3

### PR DESCRIPTION
**Pull request by zeffii**
- [x] auto self update after operator removes items (Doesn't work for the last object.. not sure why yet)
- [x] fix "Matrixes" typo in socket name
- [ ] Sort is not used for ui list display order
- [ ] maybe (want) : objs which are not connected from their verts,edges,polygons sockets, shall only do work to obtain their matrix. 
   - this might mean adding an explicit `def update` to detect a change in socket connections.. -- remember, when you disconnect outgoing sockets on node X, node X's process function is not called.

~maybe : display VEFM (Vectors, Edges, Faces, Matrix) icons~

### CODE REVIEW

**Comment by zeffii** [comment]
image

**Comment by zeffii** [comment]
was tempted to use this on a single row for the UI.. but won't.
```python
        # OR !!

        # col = layout.column(align=True)
        # row = col.row(align=True)
        # row.prop(self, 'sort', text='', toggle=True, icon='SORTALPHA')  # sort alpha
        # row.prop(self, "modifiers", text='', toggle=True, icon='MODIFIER')  # post modifier
        # row.prop(self, "vergroups", text='', toggle=True, icon='GROUP_VERTEX') # vertex groups
        # row.operator(callback, text="Select").fn_name = 'select_objs'
```

**Comment by nortikin** [comment]
but we always can reduce height a little

**Comment by zeffii** [comment]
are we ready for this? i'd like to push soon. but will test a bit with iojson first.

**Comment by nortikin** [comment]
when i try to remove objects from list, it rempoves wrong items. it seems we need rebuild list dictionary again each time

**Comment by zeffii** [comment]
Ok. i'll wait till there is time to test / fix

**Comment by nortikin** [comment]
it seems uilist count every time wrong index of item

**Comment by nortikin** [comment]
i found how to fix

File: nodes/scene/objects_mk3.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class SvOB3BDataCollection(bpy.types.PropertyGroup):
 
 class SvOB3BNamesList(bpy.types.UIList):
 
-    def draw_item(self, context, layout, data, item, icon, active_data, active_propname):
+    def draw_item(self, context, layout, data, item, icon, active_data, active_propname, index):
```
Comment (zeffii) [inline nodes/scene/objects_mk3.py]: cool. looks correct.

Comment (zeffii) [inline nodes/scene/objects_mk3.py]: maybe variable `self.active_object_index` is no longer needed then..

Comment (zeffii) [inline nodes/scene/objects_mk3.py]: oh.. i think that variable _is_needed, but wasn't being used correctly without the `index` parameter

**Comment by zeffii** [comment]
could add an operator to each list item 'hide', 'select', 'edit' .

**Pull request closed without merging**